internal/server/grpc/grpc.go

```diff
@@ -102,3 +146,36 @@ func (s *Server) Stop() error {
 
 	return nil
 }
+
+func loadCertFile(certFile string) ([]byte, error) {
+	if certFile != "" {
+		return os.ReadFile(certFile)
+	}
+	return []byte{}, nil
+}
```

Comment:
Why not just call os.ReadFile instead of having this function?
os.ReadFile will return an error when the cert is missing, which you can check directly.

Reply:
this function is used multiple times and has more than one check, using this function saves on locs